cs/cli/vowpalwabbit.cpp

```diff
@@ -764,8 +764,9 @@ size_t hashstring(String^ s, size_t u)
 Func<String^, size_t, size_t>^ VowpalWabbit::GetHasher()
 { //feature manipulation
   string hash_function("strings");
-  if (m_vw->opts_n_args.vm.count("hash"))
-  { hash_function = m_vw->opts_n_args.vm["hash"].as<string>();
+  if (m_vw->options->was_supplied("hash"))
+  {
+    hash_function = m_vw->options->get_typed_option<string>("hash").value();
```

Comment:
Have you considered removing the extra ->?

Reply:
Which `->` are you referring to?

Reply:
Ping on this.

Reply:
There is actually a situation where we need a pointer rather than a reference here. If a command line is supplied we must construct the options_i object for the user, and so we must delete when the vw object is deleted. We could could cast our reference to a pointer in this case but I think we should avoid that. Most usages of options_i in the system is through references

Reply:
I'm not quite following this.   Deleting of members seems routine?